src/main/java/io/dropwizard/revolver/resource/RevolverRequestResource.java

```diff
@@ -221,31 +225,50 @@ private Response processRequest(final String service, final RevolverHttpApiConfi
 
     private Response executeInline(final String service, final RevolverHttpApiConfig api, final RevolverHttpApiConfig.RequestMethod method,
                                    final String path, final HttpHeaders headers,
-                                   final UriInfo uriInfo, final byte[] body) throws IOException, TimeoutException {
+                                   final UriInfo uriInfo, final byte[] body)
+            throws IOException, TimeoutException, ExecutionException, RetryException {
         val sanatizedHeaders = new MultivaluedHashMap<String, String>();
         headers.getRequestHeaders().forEach(sanatizedHeaders::put);
         cleanHeaders(sanatizedHeaders, api);
         val httpCommand = RevolverBundle.getHttpCommand(service, api.getApi());
-        val response = httpCommand.execute(
-                RevolverHttpRequest.builder()
-                        .traceInfo(
-                                TraceInfo.builder()
-                                        .requestId(headers.getHeaderString(RevolversHttpHeaders.REQUEST_ID_HEADER))
-                                        .transactionId(headers.getHeaderString(RevolversHttpHeaders.TXN_ID_HEADER))
-                                        .timestamp(System.currentTimeMillis())
-                                        .build())
-                        .api(api.getApi())
-                        .service(service)
-                        .path(path)
-                        .method(method)
-                        .headers(sanatizedHeaders)
-                        .queryParams(uriInfo.getQueryParameters())
-                        .body(body)
-                        .build()
-        );
-        return transform(headers, response, api.getApi(), path, method);
+
+        RevolverHttpResponse revolverHttpResponse;
+        if(null != api.getRetryConfig() && api.getRetryConfig().isEnabled()) {
```

Comment:
@phaneesh The idea is if the call times out on one node (node being chocked), we can route the call to another node for retry

Reply:
@nitishgoyal13 -> 503 or 504 is going back from revolver (because revolver node itself is chocked).